### PR TITLE
Java: Add configuration file library.

### PR DESCRIPTION
**Pull request by sb-semmle**
@yh-semmle This is a small library to work with the extracted properties (and possibly other configuration) files. I cannot easily add tests for this code yet, as we don't extract properties by default, but I plan to enable the extraction in the auto-builder and will then add test cases in a separate PR.

### CODE REVIEW

File: java/ql/src/semmle/code/configfiles/ConfigFiles.qll [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  abstract string toString();
+}
+
+/**
+ * A name-value pair often used to store configuration properties
+ * for applications, such as the port, name or address of a database.
+ */
+class ConfigPair extends @config, ConfigLocatable {
+
+  /** If this ConfigPair has a name element, return it. */
```
Comment (yh-semmle) [inline java/ql/src/semmle/code/configfiles/ConfigFiles.qll]: Please make sure that the QLDoc adheres to the [style guide](https://github.com/Semmle/ql/blob/master/docs/ql-style-guide.md#documentation), both here and elsewhere in this file.

For example, the QLDoc above could be written as: "Gets the name component of this `ConfigPair`, if any."

Comment (aschackmull) [reply java/ql/src/semmle/code/configfiles/ConfigFiles.qll]: Also, be sure to run this through the autoformatter.  That's ctrl+shift+F in eclipse.

**Comment by sb-semmle** [comment]
I reformatted and adjusted the QL doc comments. Ideally, I would like to add some tests, but `qltest` does not yet extract `.properties` files. I'll try to teach it to do so first and then come back to this to add a test or two.

File: java/ql/src/semmle/code/configfiles/ConfigFiles.qll [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Provides classes and predicates for working with configuration files, such
+ * as Java .properties or .ini files.
+ */
+
+import semmle.code.Location
+
+/** Each element in a configuration file has a location. */
```
Comment (yh-semmle) [inline java/ql/src/semmle/code/configfiles/ConfigFiles.qll]: "An element in a configuration file that has a location."

File: java/ql/src/semmle/code/configfiles/ConfigFiles.qll [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Provides classes and predicates for working with configuration files, such
+ * as Java .properties or .ini files.
```
Comment (yh-semmle) [inline java/ql/src/semmle/code/configfiles/ConfigFiles.qll]: Backticks around `.properties` and `.ini`.

File: java/ql/src/semmle/code/configfiles/ConfigFiles.qll [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  /** Gets a textual representation of this element. */
+  abstract string toString();
+}
+
+/**
+ * A name-value pair often used to store configuration properties
+ * for applications, such as the port, name or address of a database.
+ */
+class ConfigPair extends @config, ConfigLocatable {
+  /** Gets the name of this ConfigPair, if any. */
```
Comment (yh-semmle) [inline java/ql/src/semmle/code/configfiles/ConfigFiles.qll]: Backticks around `ConfigPair` here and elsewhere.

File: java/ql/src/semmle/code/configfiles/ConfigFiles.qll [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+/** The value element of a ConfigPair. */
+class ConfigValue extends @configValue, ConfigLocatable {
+  /** Gets the value as a string. */
+  string getValue() { configValues(this, _, result) }
+
+  /** Gets a printable representation of this ConfigValue. */
+  override string toString() { result = getValue() }
+}
+
+/** A Java property is a name-value pair in a .properties file. */
```
Comment (yh-semmle) [inline java/ql/src/semmle/code/configfiles/ConfigFiles.qll]: Backticks around `.properties`.

**Pull request merged**